code/modules/power/fission/core.dm

```diff
@@ -0,0 +1,302 @@
+#define RATIO_PER_NEUTRON      0.0000001
+#define MIN_RADIATION_ENERGY        2000
+#define ACTIVE_THRESHOLD             200
+#define MAX_RODS                       7
+#define JUMPED_FLUX                 1000
+#define JUMPED_ENERGY               1200
+#define MAX_RADS                      50
+
+#define MAX_INTERACT_RATIO           0.5
+#define HI_NEUT_DIVISOR              125
+#define LO_NEUT_DIVISOR               50
+#define CONTROL_ROD_DIVISOR            5
+
+/obj/machinery/atmospherics/unary/fission_core
+	name = "nuclear fission core"
+	desc = "An advanced atomic pile used to generate heat and create isotopes via nuclear fission."
+	icon = 'icons/obj/machines/power/fission.dmi'
+	icon_state = "fission_core"
+	layer = ABOVE_HUMAN_LAYER
+	density = 1
+	stat_immune = NOINPUT | NOSCREEN
+	base_type = /obj/machinery/atmospherics/unary/fission_core
+	construct_state = /decl/machine_construction/default/panel_closed
+	connect_types = CONNECT_TYPE_REGULAR|CONNECT_TYPE_FUEL
+
+	var/control_rod_depth = 1			   // 0-1, how deeply the control rods are inserted into the core, controlling neutron absorption.
+	var/neutron_flux = 0				   // The amount of neutrons in the core. Determines the ratio of fuel inside the core that is interacted with each tick.
+	var/neutron_energy = 0			  	   // The average energy of the neutrons in the core. Df
+
+	var/list/fuel_rods = list()			   // Material inserted into the core, fuel, moderators etc.
+
+	var/max_temperature = 1000			   // Maximum temperature of the interior gas mixture before the core begins to take damage.
+	var/last_message					   // Time of the last damage warning message.
+	var/damage = 0
+	var/melted_down = FALSE
```

Comment:
Please fix the whitespace in these lines, specifically make them all use spaces. Currently it is a sin against man and God:
![image](https://user-images.githubusercontent.com/2468979/122626504-4eeed180-d0ee-11eb-9715-cb06548ad412.png)